### DISCUSSION TITLE
如何使用 chatGLM3 分析代码库？

### DISCUSSION THREAD

**Discussion by ponponon** [zai-org/ChatGLM3#464] [Q&A] [answered]
我们公司有很多历史代码，托管在私有化的 gitlab 上面，这些代码都是十几年前的人写的，相关人员都已经离职，并且使用的技术栈相比于这个时代都很陈旧。但是这些代码还在生产环境运行，还需要人维护。

现在的问题是，新招的程序员，想要去理解熟悉代码库，是一个非常痛苦和漫长的过程。所以我希望可以让 chatGLM3 分析代码库，理解这些古老代码的含义和用法。帮助新员工快速上手这些老古董代码

我有什么办法可以实现这一点？需要将这些代码都投喂给 chatGLM3 做重新训练吗？

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
感觉这个需求GLM3-6B估计比吃力
但是思路大概是这样的：
1. 你写一个脚本或者工具能访问到你们自己的代码库
2. 大模型访问这些代码库（最好是有代码能力列入codeex等模型）并分析这些代码在干啥
3. 通过大模型给新员工讲课（这中间涉及到的大模型如何转换老代码成新代码，这个非常困难）
4. 不建议微调，因为根据你的表述，你没有系统的，规则的带有标注的代码数据

**Reply by DiaoYung** [reply]
可以按照规范列出规则的带标记点代码数据给大模型学习，然后让他按照这个模式学习和编写吗？
